devito/types/petsc.py

```diff
@@ -159,7 +159,15 @@ def PETScSolve(eq, target, bcs=None, solver_parameters=None, **kwargs):
     rhs = RHSEq(b_tmp, LinearSolveExpr(eq.rhs, target=target,
                 solver_parameters=solver_parameters), subdomain=eq.subdomain)
 
-    return [matvecaction] + [rhs]
+    # Create 'mock' dependencies to ensure distinct iteration loops for each component
```

Comment:
I still struggle to understand why this is needed. Do we have a simple MFE that highlights the issue